core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1240,7 +1226,8 @@ private void deleteInternal(LockedInodePath inodePath, boolean replayed, long op
               String ufsUri = resolution.getUri().toString();
               UnderFileSystem ufs = resolution.getUfs();
               AlluxioURI parentUri = alluxioUriToDel.getParent();
-              if (!safeRecursiveUFSDeletes.containsKey(parentUri)) {
+              if (alluxioUriToDel.equals(inodePath.getUri())
```

Comment:
if `unchecked=true`, `ufsSyncChecker` always returns true => parent of directory being deleted will also return true => nothing will be deleted